### PR TITLE
Misc Fixes

### LINKED ISSUE #1067: Tilegenerator doesn't actually resize
Unity 2017.1, MRTK at 2017.1 release.

The Tilegenerator makes all the tiles, but all are the initial size.

### LINKED ISSUE #1071: Build Window improvements
Noted a few minor niggles with the Build Window:

- [x] Will let the user build with no scenes in the project (noob fail)
- [x] Make sure the selected option is reflected in the built project (test)- Unity still defaults to ARM which is not helpful
- [X] Needs to be able to support IL2CPP scripting target.  (Some Unity features such as PostProcessing only work on IL2CPP)
- [x] Update "View Log File" for local build if we're not targeting the HoloLens.
- [ ] ~~Disable / skip "Unity C# Projects" when using "Build AppX" or "Build sln / Build AppX" as this fauses the build to fail.~~

### PR DESCRIPTION
**Pull request by StephenHodgson**
Fixes #1067
Fixes #1071 
Fixes https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/1059#discussion_r142541238

image

### CODE REVIEW

**Comment by DDReaper** [comment]
Needs the extra fix I added today #1071 
Using "Unity C# projects" breaks building Appx through the custom window

**Comment by StephenHodgson** [comment]
@DDReaper let me know what you think.

File: Assets/HoloToolkit/BuildAndDeploy/Editor/BuildDeployWindow.cs [hunk 1]
```diff
@@ -672,12 +683,31 @@ private void OnGUI()
...
+                if (!remoteDeviceDetected)
+                {
+                    localLogPath = string.Format("%USERPROFILE%\\AppData\\Local\\Packages\\{0}\\TempState\\UnityPlayer.log", PlayerSettings.productName);
+                    localLogExists = File.Exists(localLogPath);
+                }
+
+                GUI.enabled = ShouldLogViewBeEnabled && (remoteDeviceDetected || isLocalBuild && localLogExists);
                 GUILayout.FlexibleSpace();
 
+
```
Comment (keveleigh) [inline Assets/HoloToolkit/BuildAndDeploy/Editor/BuildDeployWindow.cs]: nit: was adding a second empty line intentional?

Comment (StephenHodgson) [reply Assets/HoloToolkit/BuildAndDeploy/Editor/BuildDeployWindow.cs]: Nope.

**Comment by DDReaper** [comment]
Need to sleep but I also want this in too (as well as the new Master -> dev merge??) Let me quickly test

File: Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs [hunk 2]
```diff
@@ -162,7 +165,11 @@ private void UpdateSettings(BuildTarget currentBuildTarget)
                 {
                     using (var webRequest = UnityWebRequest.Get(InputManagerAssetURL))
                     {
+#if UNITY_2017_2_OR_NEWER
+                        webRequest.SendRequest();
```
Comment (DDReaper) [inline Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs]: This causes an error on my machine.  Should be:
```
webRequest.SendWebRequest();
```
using 2017.2.0.f2-MRTP build

**Comment by StephenHodgson** [comment]
@DDReaper fixed

File: Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs (hunk 2 above)
Comment (DDReaper) [inline Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs]: This causes an error on my machine.  Should be:
```
webRequest.SendWebRequest();
```
using 2017.2.0.f2-MRTP build

File: Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs [hunk 3]
```diff
@@ -102,8 +102,11 @@ private void UpdateSettings(BuildTarget currentBuildTarget)
                 {
                     using (var webRequest = UnityWebRequest.Get(SharingServiceURL))
                     {
+#if UNITY_2017_2_OR_NEWER
+                        webRequest.SendRequest();
```
Comment (DDReaper) [inline Assets/HoloToolkit/Utilities/Scripts/Editor/ProjectSettingsWindow.cs]: This causes an error on my machine.  Should be:
```
webRequest.SendWebRequest();
```
using 2017.2.0.f2-MRTP build

**Pull request merged**